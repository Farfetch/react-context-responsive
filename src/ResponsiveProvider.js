import { getBreakpoints, getMediaqueries, getQueriesObjects } from './utils';
import { useEffect } from 'react';
import PropTypes from 'prop-types';
import React, { useMemo, useRef, useState } from 'react';
import ResponsiveContext from './ResponsiveContext';

const ResponsiveProvider = ({
    initialMediaType,
    defaultOrientation,
    children,
    breakpoints,
    breakpointsMax,
    mediaQueries,
}) => {
    const [isCalculated, setIsCalculated] = useState(false);
    const [currentBreakpoint, setCurrentBreakpoint] = useState(
        initialMediaType
    );
    const [currentOrientation, setCurrentOrientation] = useState(
        defaultOrientation
    );

    const breakpointsWithInitialValue = {
        _initial: '0em',
        ...breakpoints,
    };

    const breakpointNames = Object.keys(
        mediaQueries || breakpointsWithInitialValue
    );

    if (!mediaQueries) {
        mediaQueries = getMediaqueries(
            breakpointsWithInitialValue,
            breakpointsMax,
            breakpointNames
        );
    }

    const breakpointsRef = useRef(
        getBreakpoints(mediaQueries, breakpointNames)
    );

    const queriesObjects = useMemo(
        () => getQueriesObjects(currentBreakpoint, breakpointsRef.current),
        [currentBreakpoint, breakpointsRef]
    );

    // Add event listener for breakpoints
    useEffect(() => {
        const breakpoints = breakpointsRef.current;
        const localMediaQueries = breakpoints.map(
            (breakpoint) => breakpoint.mediaQuery
        );
        const mediaQueryListsAndListeners = localMediaQueries.map(
            (mediaQuery, index) => {
                const mediaQueryList = window.matchMedia(mediaQuery);

                if (mediaQueryList.matches) {
                    setCurrentBreakpoint(breakpoints[index].mediaType);
                    setIsCalculated(true);
                }

                const listener = (event) => {
                    event.matches &&
                        setCurrentBreakpoint(breakpoints[index].mediaType);
                };

                mediaQueryList.addListener(listener);

                return [mediaQueryList, listener];
            }
        );

        return () => {
            mediaQueryListsAndListeners.forEach(
                ([mediaQueryList, listener]) => {
                    mediaQueryList.removeListener(listener);
                }
            );
        };
    }, [breakpointsRef]);

    // Add orientation media query
    useEffect(() => {
        const setMatchedOrientation = (matches) => {
            setCurrentOrientation(matches ? 'portrait' : 'landscape');
        };

        const orientationMediaQueryList = window.matchMedia(
            '(orientation: portrait)'
        );

        setMatchedOrientation(orientationMediaQueryList.matches);

        const listener = (event) => {
            setMatchedOrientation(event.matches);
        };

        orientationMediaQueryList.addListener(listener);

        return () => {
            orientationMediaQueryList.removeListener(listener);
        };
    }, []);

    const contextObject = useMemo(
        () => ({
            mediaType: currentBreakpoint,
            orientation: currentOrientation,
            isCalculated,
            ...queriesObjects,
        }),
        [currentBreakpoint, currentOrientation, isCalculated, queriesObjects]
    );

    useEffect(() => {
        if (process && process.env && process.env.NODE_ENV === 'development') {
            /* eslint-disable no-console */
            console.group(
                '%c @farfetch/react-context-responsive updated!',
                'color:#c4aa84; font-weight:bold;'
            );
            console.log(
                '%c Current breakpoint:',
                'font-weight:bold;',
                currentBreakpoint
            );
            console.log('%c Context:', 'font-weight:bold;', contextObject);
            console.groupEnd();
            /* eslint-enable no-console */
        }
    }, [contextObject, currentBreakpoint]);

    return (
        <ResponsiveContext.Provider value={contextObject}>
            {children}
        </ResponsiveContext.Provider>
    );
};

const breakpointsPropTypes = {
    xs: PropTypes.string,
    sm: PropTypes.string,
    md: PropTypes.string,
    lg: PropTypes.string,
    xl: PropTypes.string,
};

ResponsiveProvider.defaultProps = {
    initialMediaType: 'xs',
    defaultOrientation: null,
};

ResponsiveProvider.propTypes = {
    initialMediaType: PropTypes.oneOf([
        '_initial',
        'xs',
        'sm',
        'md',
        'lg',
        'xl',
    ]),
    defaultOrientation: PropTypes.oneOf(['landscape', 'portrait']),
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
    breakpoints: PropTypes.exact(breakpointsPropTypes),
    breakpointsMax: PropTypes.exact(breakpointsPropTypes),
    mediaQueries: PropTypes.exact({
        ...breakpointsPropTypes,
        _initial: PropTypes.string,
    }),
};

export default ResponsiveProvider;
