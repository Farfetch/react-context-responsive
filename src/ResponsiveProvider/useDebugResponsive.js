import { useEffect } from 'react';

const useDebugResponsive = (contextObject, currentMediaType) => {
    useEffect(() => {
        if (
            typeof process !== 'undefined' &&
            process.env &&
            process.env.NODE_ENV === 'development'
        ) {
            /* eslint-disable no-console */
            console.group(
                '%c @farfetch/react-context-responsive updated!',
                'color:#c4aa84; font-weight:bold;'
            );
            console.log(
                '%c Current media type:',
                'font-weight:bold;',
                currentMediaType
            );
            console.log('%c Context:', 'font-weight:bold;', contextObject);
            console.groupEnd();
            /* eslint-enable no-console */
        }
    }, [contextObject, currentMediaType]);
};

export default useDebugResponsive;
