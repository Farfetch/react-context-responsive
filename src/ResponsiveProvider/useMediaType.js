import { useEffect, useState } from 'react';

const useMediaType = (breakpointsRef, initialMediaType) => {
    const [isCalculated, setIsCalculated] = useState(false);
    const [currentMediaType, setCurrentMediaType] = useState(initialMediaType);

    useEffect(() => {
        const breakpoints = breakpointsRef.current;
        const localMediaQueries = breakpoints.map(
            (breakpoint) => breakpoint.mediaQuery
        );
        const mediaQueryListsAndListeners = localMediaQueries.map(
            (mediaQuery, index) => {
                const mediaQueryList = window.matchMedia(mediaQuery);

                if (mediaQueryList.matches) {
                    setCurrentMediaType(breakpoints[index].mediaType);
                    setIsCalculated(true);
                }

                const listener = (event) => {
                    event.matches &&
                        setCurrentMediaType(breakpoints[index].mediaType);
                };

                mediaQueryList.addListener(listener);

                return [mediaQueryList, listener];
            }
        );

        return () => {
            mediaQueryListsAndListeners.forEach(
                ([mediaQueryList, listener]) => {
                    mediaQueryList.removeListener(listener);
                }
            );
        };
    }, [breakpointsRef]);

    return { isCalculated, currentMediaType };
};

export default useMediaType;
