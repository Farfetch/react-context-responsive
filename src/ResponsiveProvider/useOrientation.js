import { useEffect, useState } from 'react';

const useOrientation = (defaultOrientation) => {
    const [currentOrientation, setCurrentOrientation] = useState(
        defaultOrientation
    );

    useEffect(() => {
        const setMatchedOrientation = (matches) => {
            setCurrentOrientation(matches ? 'portrait' : 'landscape');
        };

        const orientationMediaQueryList = window.matchMedia(
            '(orientation: portrait)'
        );

        setMatchedOrientation(orientationMediaQueryList.matches);

        const listener = (event) => {
            setMatchedOrientation(event.matches);
        };

        orientationMediaQueryList.addListener(listener);

        return () => {
            orientationMediaQueryList.removeListener(listener);
        };
    }, []);

    return currentOrientation;
};

export default useOrientation;
