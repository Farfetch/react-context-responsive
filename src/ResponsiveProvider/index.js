import { getBreakpoints, getMediaqueries, getQueriesObjects } from '../utils';
import PropTypes from 'prop-types';
import React, { useMemo, useRef } from 'react';
import ResponsiveContext from '../ResponsiveContext';
import useDebugResponsive from './useDebugResponsive';
import useMediaType from './useMediaType';
import useOrientation from './useOrientation';

const ResponsiveProvider = ({
    initialMediaType,
    defaultOrientation,
    children,
    breakpoints,
    breakpointsMax,
    mediaQueries,
    mobileBreakpoint,
}) => {
    const breakpointsWithInitialValue = {
        _initial: '0em',
        ...breakpoints,
    };

    const breakpointNames = Object.keys(
        mediaQueries || breakpointsWithInitialValue
    );

    if (!mediaQueries) {
        mediaQueries = getMediaqueries(
            breakpointsWithInitialValue,
            breakpointsMax,
            breakpointNames
        );
    }

    const breakpointsRef = useRef(
        getBreakpoints(mediaQueries, breakpointNames)
    );

    const { currentMediaType, isCalculated } = useMediaType(
        breakpointsRef,
        initialMediaType
    );

    const queriesObjects = useMemo(
        () => getQueriesObjects(currentMediaType, breakpointsRef.current),
        [currentMediaType, breakpointsRef]
    );

    const currentOrientation = useOrientation(defaultOrientation);

    const contextObject = useMemo(
        () => ({
            mediaType: currentMediaType,
            orientation: currentOrientation,
            isCalculated,
            mobileBreakpoint,
            ...queriesObjects,
        }),
        [
            currentMediaType,
            currentOrientation,
            isCalculated,
            mobileBreakpoint,
            queriesObjects,
        ]
    );

    useDebugResponsive(contextObject, currentMediaType);

    return (
        <ResponsiveContext.Provider value={contextObject}>
            {children}
        </ResponsiveContext.Provider>
    );
};

const breakpointsPropTypes = {
    xs: PropTypes.string,
    sm: PropTypes.string,
    md: PropTypes.string,
    lg: PropTypes.string,
    xl: PropTypes.string,
};

ResponsiveProvider.defaultProps = {
    initialMediaType: 'xs',
    defaultOrientation: null,
    mobileBreakpoint: 'md',
};

ResponsiveProvider.propTypes = {
    initialMediaType: PropTypes.oneOf([
        '_initial',
        'xs',
        'sm',
        'md',
        'lg',
        'xl',
    ]),
    defaultOrientation: PropTypes.oneOf(['landscape', 'portrait']),
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
    breakpoints: PropTypes.exact(breakpointsPropTypes),
    breakpointsMax: PropTypes.exact(breakpointsPropTypes),
    mediaQueries: PropTypes.exact({
        ...breakpointsPropTypes,
        _initial: PropTypes.string,
    }),
    mobileBreakpoint: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
};

export default ResponsiveProvider;
