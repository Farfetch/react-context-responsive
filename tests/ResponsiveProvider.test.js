import { act, cleanup, render } from '@testing-library/react';
import { breakpoints, breakpointsMax } from './__fixtures__/mockBreakpoints';
import {
    mockMatchMedia,
    unmockMatchMedia,
} from './__fixtures__/mockMatchMedia';
import React from 'react';
import ResponsiveProvider from '../src/ResponsiveProvider';
import mockContextContent from './__fixtures__/mockContextContent';

const props = { breakpoints, breakpointsMax };

jest.mock('../src/ResponsiveContext');

global.console = {
    log: jest.fn(),
    group: jest.fn(),
    groupEnd: jest.fn(),
    error: jest.fn(),
};

describe('<ResponsiveProvider />', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        mockMatchMedia();
    });

    afterEach(() => {
        cleanup();
        unmockMatchMedia();
    });

    test.each`
        width   | height
        ${240}  | ${320}
        ${320}  | ${240}
        ${375}  | ${812}
        ${812}  | ${375}
        ${320}  | ${568}
        ${768}  | ${1024}
        ${1024} | ${768}
        ${1024} | ${1366}
        ${1336} | ${1024}
        ${1440} | ${900}
        ${2560} | ${1440}
    `(
        'should pass to the context the correct value when the width is $width and the height is $height, after calculation',
        ({ width, height }) => {
            window.resizeTo(width, height);

            render(
                <ResponsiveProvider {...props}>
                    <div>Responsive test</div>
                </ResponsiveProvider>
            );

            const { default: Context } = require('../src/ResponsiveContext');

            // One time with default values, the other with the calculated
            expect(Context.Provider).toHaveBeenCalledTimes(2);

            const passedValues = Context.Provider.mock.calls[1][0].value;

            expect(passedValues).toMatchSnapshot();
        }
    );

    test('should pass to the context the default values, before calculation', () => {
        render(
            <ResponsiveProvider {...props}>
                <div>Responsive test</div>
            </ResponsiveProvider>
        );

        const { default: Context } = require('../src/ResponsiveContext');
        const passedValues = Context.Provider.mock.calls[0][0].value;

        expect(passedValues).toMatchSnapshot();
    });

    test('should pass to the context the props values, before calculation, if available', () => {
        render(
            <ResponsiveProvider
                {...props}
                initialMediaType={'xl'}
                defaultOrientation={'portrait'}
            >
                <div>Responsive test</div>
            </ResponsiveProvider>
        );

        const { default: Context } = require('../src/ResponsiveContext');
        const passedValues = Context.Provider.mock.calls[0][0].value;

        expect(passedValues).toMatchSnapshot();
    });

    test('should pass to the context the correct values, on resize', () => {
        window.resizeTo(1024, 768);
        render(
            <ResponsiveProvider {...props}>
                <div>Responsive test</div>
            </ResponsiveProvider>
        );

        act(() => {
            window.resizeTo(1440, 900);
        });

        const { default: Context } = require('../src/ResponsiveContext');

        const passedValues = Context.Provider.mock.calls[1][0].value;

        expect(passedValues).toMatchSnapshot();
    });

    test('should remove all the added matchMedia listeners on unmount', () => {
        window.resizeTo(1024, 768);

        const { unmount } = render(
            <ResponsiveProvider {...props}>
                <div>Responsive test</div>
            </ResponsiveProvider>
        );

        const listenersAdded = window._mockMatchMedia.reduce(
            (memo, mockMatchMedia) => {
                return memo + mockMatchMedia.addListener.mock.calls.length;
            },
            0
        );
        const listenersRemoved = window._mockMatchMedia.reduce(
            (memo, mockMatchMedia) => {
                return memo + mockMatchMedia.removeListener.mock.calls.length;
            },
            0
        );

        // All the 6 breakpoints + 1 orientation
        expect(listenersAdded).toBe(7);
        expect(listenersRemoved).toBe(0);

        unmount();

        const listenersRemovedAfterUnmount = window._mockMatchMedia.reduce(
            (memo, mockMatchMedia) => {
                return memo + mockMatchMedia.removeListener.mock.calls.length;
            },
            0
        );

        expect(listenersRemovedAfterUnmount).toBe(listenersAdded);
    });

    test.each(
        ['production', 'test'],
        'should not call console.log in the % environment',
        (env) => {
            process.env.NODE_ENV = env;
            window.resizeTo(1024, 768);

            render(
                <ResponsiveProvider {...props}>
                    <div>Test</div>
                </ResponsiveProvider>
            );

            expect(global.console.group).not.toHaveBeenCalled();
            expect(global.console.log).not.toHaveBeenCalled();
            expect(global.console.groupEnd).not.toHaveBeenCalled();

            // Clean the global changes
            process.env.NODE_ENV = 'test';
        }
    );

    test('should call console.log only for development environment', () => {
        process.env.NODE_ENV = 'development';
        window.resizeTo(1024, 768);

        const { unmount } = render(
            <ResponsiveProvider {...props}>
                <div>Test</div>
            </ResponsiveProvider>
        );

        expect(global.console.group).toHaveBeenCalledTimes(2);
        expect(global.console.group).toHaveBeenCalledWith(
            '%c @farfetch/react-context-responsive updated!',
            'color:#c4aa84; font-weight:bold;'
        );
        expect(global.console.log).toHaveBeenCalledTimes(4);
        expect(global.console.log).toHaveBeenCalledWith(
            '%c Context:',
            'font-weight:bold;',
            mockContextContent
        );
        expect(global.console.groupEnd).toHaveBeenCalledTimes(2);

        // Clean the global changes
        process.env.NODE_ENV = 'test';
        unmount();
    });
});
