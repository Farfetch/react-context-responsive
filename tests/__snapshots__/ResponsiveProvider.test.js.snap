// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 240 and the height is 320, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "is": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": true,
    "xs": true,
  },
  "mediaType": "_initial",
  "orientation": "portrait",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 320 and the height is 240, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": true,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": true,
    "xs": false,
  },
  "mediaType": "xs",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 320 and the height is 568, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": true,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": true,
    "xs": false,
  },
  "mediaType": "xs",
  "orientation": "portrait",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 375 and the height is 812, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": true,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": true,
    "xs": false,
  },
  "mediaType": "xs",
  "orientation": "portrait",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 768 and the height is 1024, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": true,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "sm",
  "orientation": "portrait",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 812 and the height is 375, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": true,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "sm",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 1024 and the height is 768, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": true,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "md",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 1024 and the height is 1366, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": true,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "md",
  "orientation": "portrait",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 1336 and the height is 1024, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": true,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": true,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "lg",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 1440 and the height is 900, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": true,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": true,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "lg",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct value when the width is 2560 and the height is 1440, after calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "mediaType": "xl",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the correct values, on resize 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": true,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "isCalculated": true,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "mediaType": "md",
  "orientation": "landscape",
}
`;

exports[`<ResponsiveProvider /> should pass to the context the default values, before calculation 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": true,
  },
  "isCalculated": false,
  "lessThan": Object {
    "_initial": false,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": true,
    "xs": false,
  },
  "mediaType": "xs",
  "orientation": null,
}
`;

exports[`<ResponsiveProvider /> should pass to the context the props values, before calculation, if available 1`] = `
Object {
  "greaterThan": Object {
    "_initial": true,
    "lg": true,
    "md": true,
    "sm": true,
    "xl": false,
    "xs": true,
  },
  "is": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": true,
    "xs": false,
  },
  "isCalculated": false,
  "lessThan": Object {
    "_initial": false,
    "lg": false,
    "md": false,
    "sm": false,
    "xl": false,
    "xs": false,
  },
  "mediaType": "xl",
  "orientation": "portrait",
}
`;
